src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WSTrustChannel.cs

```diff
@@ -0,0 +1,427 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IdentityModel.Tokens;
+using System.ServiceModel.Channels;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.IdentityModel.Protocols.WsTrust;
+using Microsoft.IdentityModel.Tokens;
+using WCFSecurityToken = System.IdentityModel.Tokens.SecurityToken;
+
+namespace System.ServiceModel.Federation
+{
+    /// <summary>
+    /// A channel that is used to send WS-Trust messages to an STS.
+    /// </summary>
+    public class WSTrustChannel : IWSTrustChannelContract, IChannel, ICommunicationObject
+    {
+        private const int FaultMaxBufferSize = 20 * 1024;
+
+        /// <summary>
+        /// Constructs a <see cref="WSTrustChannel" />.
+        /// </summary>
+        /// <param name="channelfactory">The <see cref="ChannelFactory" /> that is creating this object.</param>
+        /// <param name="trustVersion">The version of WS-Trust this channel will use for serializing <see cref="Message" /> objects.</param>
+        public WSTrustChannel(ChannelFactory channelFactory,
+                              WsTrustVersion trustVersion)
+        {
+            if (!(channelFactory is IRequestChannel requestChannel))
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.GetResourceString(SR.ChannelFactoryMustSupportIRequestChannel)));
+
+            Initialize(channelFactory, requestChannel, trustVersion);
+        }
+
+        /// <summary>
+        /// Constructs a <see cref="WSTrustChannel" />.
+        /// </summary>
+        /// <param name="channelfactory">The <see cref="Federation.ChannelFactory" /> that is creating this object.</param>
+        /// <param name="requestChannel">The <see cref="IRequestChannel" /> this object will be used to send and receive <see cref="Message" /> objects.</param>
```

Comment:
Should the cref say Federation.ChannelFactory? I thought this was a regular S.SM.ChannelFactory?

Reply:
address in: ab9b3c1cb00f558dcd6ac6f3b02663ace5e25750